finch/processes/wps_geoseries_to_netcdf.py

```diff
@@ -0,0 +1,200 @@
+                ],
+            ),
+            wpsio.output_log,
+        ]
+
+        super().__init__(
+            self._handler,
+            identifier="geoseries_to_netcdf",
+            version="0.1",
+            title="Convert a geospatial series to a CF netCDF.",
+            abstract="Reshapes a geospatial series with a features dimensions and converts into the netCDf format, following CF convetions.",
+            inputs=inputs,
+            outputs=outputs,
+            status_supported=True,
+            store_supported=True,
+        )
+
+        self.status_percentage_steps = {
+            "start": 5,
+            "downloaded": 25,
+            "converted": 75,
+            "done": 99,
+        }
+
+    def _handler(self, request, response):
+        write_log(self, "Processing started", process_step="start")
+
+        # --- Process inputs ---
+        geo_url = request.inputs["resource"][0].url
+        index_dim = request.inputs['index_dim'][0].data
+        feat_dim = request.inputs['feat_dim'][0].data
+        squeeze = request.inputs['squeeze'][0].data
+        grid_map = request.inputs['grid_mapping'][0].data
+
+        # Open URL
+        gdf = gpd.read_file(geo_url)
+        write_log(self, "Geoseries downloaded", process_step="downloaded")
+
+        # Try casting all columns
+        gdf = _maybe_cast(gdf)
+
+        # Set index and convert to xr
+        ds = gdf.set_index(index_dim).to_xarray()
+
+        # Reshape geometries
+        ds = cfgeo.reshape_unique_geometries(ds)
+
+        # Convert shapely objects to CF-style
+        coords = cfgeo.shapely_to_cf(ds.geometry, grid_mapping=grid_map)
+        if coords.features.size == coords.node.size:
+            # Then it's only single points, we can drop 'node'
+            coords = coords.drop_dims('node')
+
+        ds = xr.merge([ds.drop_vars('geometry'), coords])
+
+        # feat dim
+        if feat_dim:
+            feat = _maybe_squeeze(ds[feat_dim], index_dim)
+            if feat.ndim == 2:
+                raise ValueError("'feat_dim' was given but it cannot be squeezed along the index dimension.")
+            ds[feat_dim] = feat
+            ds = ds.drop_vars('features').rename(features=feat_dim).set_coords(feat_dim)
+
+        # squeeze
+        if squeeze:
+            for dim in [feat_dim, index_dim]:
+                for name, var in ds.data_vars.items():
+                    if dim in var.dims:
+                        try:
+                            ds[name] = _maybe_squeeze(var, dim)
+                        except KeyError:
+                            print(ds)
+                            raise
+
+        write_log(self, "Geoseries converted to CF dataset", process_step="converted")
+
+        host = urlparse(geo_url).netloc
+        if host is None:
+            source = "geospatial series data"
+        else:
+            source = f"data downloaded from {host}"
+        ds.attrs["history"] = update_history(
```

Comment:
Add timestamp.

Reply:
`update_history` already does that! I also specifies the xclim version, but we are not using xclim here... Is this a problem? Should we reimplement a version for finch?

Reply:
oh, right... 

If this is going to go in prod for ECCC, then yes.